system-design/class-diagram/domain/entity.puml

```diff
@@ -0,0 +1,48 @@
+@startuml
+package entity {
+
+    class Book {
+        + Id: Guid
+        + Title: string
+        + Author: string
+        + Isbn: long
+        + Genre: string
+        + PublicationYear: DateTime
+        + ToString(): string
+    }
+
+    class Patron {
+        + Id: Guid
+        + Name: string
+        + MembershipNumber: long
+        + ContactDetails: long
+        + ToString(): string
+    }
+
+    class Borrow {
+        + Id: Guid
+        + PatronId: Guid
+        + BookId: Guid
+        + Status: BorrowStatus
+        + DueDate: DateTime
+        + BorrowDate: DateTime
+        + ToString(): string
+    }
+
+    interface IEntity {
+        + Id: Guid
+    }
+    IEntity <|.. Borrow
+    IEntity <|.. Book
+    IEntity <|.. Patron
+
+    enum BorrowStatus {
+        Borrowed
+        Overdue
+        Returned
+        Reserved
+    }
+    BorrowStatus *-- Borrow
```

Comment:
I think that this is not composition, is just **"Directed association"**.

Reply:
I decided for composition because if the `Borrow` entity is removed from the system, `BorrowStatus` loses its meaning.